### PR TITLE
Fix ctc_batch_cost() error when batch_size = 1

### PR DESCRIPTION
**Pull request by WindQAQ**
Fix [issue #7049](https://github.com/keras-team/keras/issues/7049):
When batch size is 1, 
```
label_length = tf.to_int32(tf.squeeze(label_length))
input_length = tf.to_int32(tf.squeeze(input_length))
```
squeeze out all axes so that `label_length` and `input_length` become tensors with shape (). The functions will crush because of the shape error. Hence, just squeezing the last axis to keep the tensors with shape (samples, ).

### CODE REVIEW

File: keras/backend/tensorflow_backend.py [hunk 1]
```diff
@@ -3942,8 +3942,8 @@ def ctc_batch_cost(y_true, y_pred, input_length, label_length):
         Tensor with shape (samples,1) containing the
             CTC loss of each element.
     """
-    label_length = tf.to_int32(tf.squeeze(label_length))
-    input_length = tf.to_int32(tf.squeeze(input_length))
+    label_length = tf.to_int32(tf.squeeze(label_length, axis=1))
+    input_length = tf.to_int32(tf.squeeze(input_length, axis=1))
```
Comment (fchollet) [inline keras/backend/tensorflow_backend.py]: `axis=1` is the first axis, not the last axis (-1).

Please add a unit test.

Comment (WindQAQ) [reply keras/backend/tensorflow_backend.py]: @fchollet 
Both original inputs `label_length` and `input_length` are **2D** tensors with shape (samples, 1) and the index of `axis` starts at 0, so I think either `axis=1` or `axis=-1` refers to the same axis.

Comment (fchollet) [reply keras/backend/tensorflow_backend.py]: Sure, but for the sake of semantics and possible future extensibility, please use `-1` to refer to the last axis.

Comment (WindQAQ) [reply keras/backend/tensorflow_backend.py]: Okay.
By the way, should I add a unit test at `test_ctc()` in `backend_test.py`, right?

Comment (fchollet) [reply keras/backend/tensorflow_backend.py]: You can add a test case in https://github.com/keras-team/keras/blob/master/tests/keras/backend/backend_test.py#L1446

Or you can just add a new function.

**Comment by WindQAQ** [comment]
Change `axis=1` to `axis=-1`. The test case with one sample only is added

**Comment by WindQAQ** [comment]
Change axis=1 to axis=-1. The test case with one sample only is added
